plugins/BTagAnalyzer.cc

```diff
@@ -1796,20 +1796,20 @@ void BTagAnalyzerT<IPTI,VTX>::processJets(const edm::Handle<PatJetCollection>& j
     if ( runFatJets_ && iJetColl == 0 )
     {
       // N-subjettiness
-      JetInfo[iJetColl].Jet_tau1[JetInfo[iJetColl].nJet] = pjet->userFloat("Njettiness:tau1");
-      JetInfo[iJetColl].Jet_tau2[JetInfo[iJetColl].nJet] = pjet->userFloat("Njettiness:tau2");
+      JetInfo[iJetColl].Jet_tau1[JetInfo[iJetColl].nJet] = ( pjet->hasUserFloat("Njettiness:tau1") ? pjet->userFloat("Njettiness:tau1") : pjet->userFloat("NjettinessAK8:tau1") );
+      JetInfo[iJetColl].Jet_tau2[JetInfo[iJetColl].nJet] = ( pjet->hasUserFloat("Njettiness:tau2") ? pjet->userFloat("Njettiness:tau2") : pjet->userFloat("NjettinessAK8:tau2") );
       // SoftDrop kinematics
-      JetInfo[iJetColl].Jet_ptSoftDrop[JetInfo[iJetColl].nJet]    = pjet->userFloat("SoftDrop:Pt");
-      JetInfo[iJetColl].Jet_etaSoftDrop[JetInfo[iJetColl].nJet]   = pjet->userFloat("SoftDrop:Eta");
-      JetInfo[iJetColl].Jet_phiSoftDrop[JetInfo[iJetColl].nJet]   = pjet->userFloat("SoftDrop:Phi");
-      JetInfo[iJetColl].Jet_massSoftDrop[JetInfo[iJetColl].nJet]  = pjet->userFloat("SoftDrop:Mass");
-      JetInfo[iJetColl].Jet_jecF0SoftDrop[JetInfo[iJetColl].nJet] = pjet->userFloat("SoftDrop:jecFactor0");
+      JetInfo[iJetColl].Jet_ptSoftDrop[JetInfo[iJetColl].nJet]    = ( pjet->hasUserFloat("SoftDrop:Pt")         ? pjet->userFloat("SoftDrop:Pt")         : 0. );
+      JetInfo[iJetColl].Jet_etaSoftDrop[JetInfo[iJetColl].nJet]   = ( pjet->hasUserFloat("SoftDrop:Eta")        ? pjet->userFloat("SoftDrop:Eta")        : 0. );
+      JetInfo[iJetColl].Jet_phiSoftDrop[JetInfo[iJetColl].nJet]   = ( pjet->hasUserFloat("SoftDrop:Phi")        ? pjet->userFloat("SoftDrop:Phi")        : 0. );
```

Comment:
Because they are not stored in MiniAOD. Btw, thanks for the question. I now realized there is a copy/paste mistake in the pruned mass. I will fix this.